pkg/backends/otlp/internal/data/event.go

```diff
@@ -0,0 +1,64 @@
+package data
+
+import (
+	"errors"
+	"time"
+
+	v1common "go.opentelemetry.io/proto/otlp/common/v1"
+	v1log "go.opentelemetry.io/proto/otlp/logs/v1"
+
+	"github.com/atlassian/gostatsd"
+)
+
+type OtlpEvent struct {
+	raw *gostatsd.Event
+}
+
+func NewOtlpEvent(e *gostatsd.Event) (*OtlpEvent, error) {
+	if e == nil {
+		return nil, errors.New("event not found")
+	}
+	return &OtlpEvent{raw: e}, nil
+}
+
+func (s *OtlpEvent) TransformToLog() *v1log.LogRecord {
+	attrs := make(map[string]string)
+	e := s.raw
+
+	dimensions := e.Tags.ToMap()
+	for key, value := range dimensions {
```

Comment:
Should these still be inside a map called tags i.e. so the gateway processor can distinguish these dimensions from those below as a tags?

Reply:
Currently I'm flattening it and override the tags with fields on the events. As tested here https://github.com/atlassian/gostatsd/pull/694/files#diff-8f2c078c61ccc41bce7f801d20bd30203625ca465ca38f0ea9c1afed0d3df7b0R66

I think you're right, we should let downstream decide how to deal with it.